pkg/operator/ceph/csi/secrets.go

```diff
@@ -183,38 +237,199 @@ func CreateCSISecrets(context *clusterd.Context, clusterInfo *client.ClusterInfo
+
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		cephCluster.Status.Cephx.CSI.CephxStatus = &cephxStatus
+		cephCluster.Status.Cephx.CSI.PriorKeyCount = uint8(currentKeyCount) //nolint:gosec // disable G115
+
+		if err := reporting.UpdateStatus(context.Client, cephCluster); err != nil {
+			return errors.Wrapf(err, "failed to update cephCluster %q to update csi cephx status to %q in namespace %q", namespacedName.Name, cephxStatus, namespacedName.Namespace)
+		}
+		return nil
+	})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get and update Ceph cluster %q status in namespace %q when updating csi cephx status", namespacedName.Name, namespacedName.Namespace)
+	}
+
+	logger.Debugf("successfully updated Ceph cluster %q status with CSI Cephx status in namespace %q", namespacedName.Name, namespacedName.Namespace)
+	return nil
+}
+
+// getCsiKeyRotationInfo runs the `ceph auth ls` command to fetch all the keys and filter out the keys with same base name. Example
+// for base name `client.csi-rbd-node`.
+// From the list of key name return from `getMatchingClient` we'll read the suffix index eg; for key `client.csi-rbd-node.3` we'll take `3` and compare with
+// KeyGeneration and basaed on the comparison we'll return bool to check if we should rotate the keys or not and returning list
+// of all the matching keys.
+func getCsiKeyRotationInfo(context *clusterd.Context, clusterInfo *client.ClusterInfo, keyBaseName string) (int, []string, error) {
+	authList, err := client.AuthList(context, clusterInfo)
+	if err != nil {
+		return 0, []string{}, err
+	}
+
+	keyWithBaseName, err := getMatchingClient(authList, keyBaseName)
+	if err != nil {
+		return 0, keyWithBaseName, err
+	}
+
+	// In fresh deployment `keyWithBaseName` length will be empty, so let's not return error.
+	if len(keyWithBaseName) == 0 {
+		logger.Debugf("no key matching with %s found in auth list must be fresh cluster", keyBaseName)
+		return 0, keyWithBaseName, nil
+	}
+
+	sortedClientList, err := sortCSIClientName(keyWithBaseName)
+	if err != nil {
+		return 0, keyWithBaseName, errors.Wrapf(err, "failed to get sorted CSI key list from %s while reading `maxCount`", sortedClientList)
+	}
+
+	if len(sortedClientList) == 0 {
+		return 0, keyWithBaseName, fmt.Errorf("empty list found while fetching CSI key from")
+	}
+
+	_, currentMaxKeyGen, err := parseCsiClient(sortedClientList[len(sortedClientList)-1])
+	if err != nil {
+		return 0, keyWithBaseName, errors.Wrap(err, "failed to get currentMaxKeyCount from key entity")
+	}
+
+	return currentMaxKeyGen, keyWithBaseName, nil
+}
+
+// getMatchingClient return list of all the ceph key which contains `client.csi-rbd-node`, we will have keys like
+// `client.csi-rbd-node.1`, `client.csi-rbd-node.2`, `client.csi-rbd-node.3` so the list will have all these key name.
```

Comment:
Does this comment needs updated to reflect that `client.` is being stripped from the front of each returned item?

Also please indicate whether `clientBaseName` is expected to contain the `client` prefix or not